packages/core/src/link/Link.tsx

```diff
@@ -39,6 +39,9 @@ export const Link = forwardRef<HTMLAnchorElement, LinkProps>(function Link(
     css: linkCss,
     window: targetWindow,
   });
+  const { ExternalIcon } = useIcon();
+
+  const LinkIconComponent = IconComponent ?? ExternalIcon;
```

Comment:
This is a breaking change, it should be:

```suggestion
  const LinkIconComponent = IconComponent === undefined ? ExternalIcon : IconComponent;
```
to match the behaviour of a default value. Right now this breaks IconComponent={null}

https://28d5e102.saltdesignsystem-storybook.pages.dev/?path=/story/core-link--target-blank-no-icon&pr=4002

Reply:
We should add a test for this use-case

Reply:
added test and applied suggestion